### DISCUSSION TITLE
如何参与到LogicFlow

### DISCUSSION THREAD

**Discussion by wumail** [didi/LogicFlow#1300] [Announcements]
> LogicFlow对外持开放态度，不论是只修改文档的标点符号还是对LogicFlow的整体功能有大的重构我们都欢迎。

## 编写LogicFlow的文档

1. 如果您在使用LogicFlow的过程中，发现文档有问题，可以直接点击左下角的**在 Github 上改善此页**来对LogicFlow的文档进行修改。此链接会自动跳转到github，并为您提供一个修改此文件的页面。

<img width="400" alt="image" src="https://user-images.githubusercontent.com/4093984/181867138-c0e81bbf-83b1-4445-a832-0f804cb78745.png">

2. 您还可以通过将LogicFlow的源代码下载到本地后，通过命名`npm run docs:dev`在本地允许LogicFlow的文档。更多的本地代码如何运行请查看[developer.md](https://github.com/didi/LogicFlow/blob/master/developer.md)。

## 给LogicFlow贡献插件

LogicFlow为了保证足够的灵活性，所以基于[插件化思想](http://logic-flow.org/article/article02.html#logicflow-%E7%9A%84%E6%8B%93%E5%B1%95%E8%83%BD%E5%8A%9B)开发方式，将很多功能都放到了`@logicflow/extension`包中。LogicFlow的插件功能非常灵活，我们即支持您将流程图上面的某个组件封装为插件, 例如[菜单](https://github.com/didi/LogicFlow/blob/master/packages/extension/src/components/menu/index.ts); 也支持您把整个流程图封装为插件, 例如[BpmnElement](https://github.com/didi/LogicFlow/blob/master/packages/extension/src/bpmn/index.ts)； 甚至你还可以把多个插件组合到一起, 将一个完整的流程图产品封装为插件。

我们期待您把您的插件贡献出来，你可以选择如下方式：

1. 将您的插件放到LogicFlow源码的extension目录下，通过Pull Request的方式提交给我们，我们会在后面统一将你的插件发布到`@logicflow/extenion`包中。
2. 将您的插件独立发布，然后将如何使用您的插件，以及插件的使用效果写到LogicFlow文档中。

## 分享LogicFlow的使用案例

LogicFlow为了提供足够的扩展性，默认情况下并不会做过多的样式、交互上的效果。我们提供了足够多的API支持您去实现您们产品、UI的效果。 在不涉及公司信息安全的情况下，欢迎您把LogicFlow在您们项目中的实际效果在[LogicFlow案例分享](https://github.com/didi/LogicFlow/issues/716)的issue中分享出来。

(PS: copied from https://github.com/didi/LogicFlow/issues/717)

**Comment by liaodalin19903** [comment]
logicflow[插件化思想地址](https://site.logic-flow.cn/article/decoding-logicflows-extension-mechanism)。
